_episodes/06-testing.md

```diff
@@ -47,67 +48,70 @@ Although not essential it is helpful to have an understanding of:
 
 ### Element 1: Unit, integration and regression tests
 
-- [Unit tests](https://en.wikipedia.org/wiki/Unit_testing) isolate each part of the program and show that the individual parts are correct.
+- [Unit tests](https://en.wikipedia.org/wiki/Unit_testing) 
+or [Integration tests](https://en.wikipedia.org/wiki/Integration_testing):
+  - work on isolated parts (units) or group of units (integretion tests),
+  - verify that software operate correctly in various scenarios,
+  - usually compare observed results to well known expected results.
 
-- [Integration tests](https://en.wikipedia.org/wiki/Integration_testing) combine individual software modules and test as a group. 
+- [Regression tests](https://en.wikipedia.org/wiki/Regression_testing):
+  - verify that software previously developed and tested still performs correctly even after it 
+    was changed or interfaced with other software
+  - you don't have to knows the expected result, the assumption is that the past results were correct.
 
-- [Regression tests](https://en.wikipedia.org/wiki/Regression_testing) verify that software previously developed and tested still 
-  performs correctly even after it was changed or interfaced with other software. 
-  For regression tests you don't have to knows what the expected result should be, 
-  the assumption is that the past results were correct. 
 
+In science, regression tests perform a special role. 
+Scientists often don't know what is the correct answer before doing the research, 
+therefore results can't be compared to the priory known results.
+However, you still want to be sure that your published results are not sensitive to 
+an operating system, a specific version of library, etc.
+If your results differ when you're changing computer environment, you should understand the 
+source of the changes. 
+The first step would be automatically check the results with various environments
+whenever you make any changes.
+You can achive this by combining your regression tests with a CI platform.
 
-- Resources: More about unit, integration and regression test you might find at the software carpentry [lesson](http://katyhuff.github.io/python-testing/index.html).
+Regression tests (as well as unit and any other type of tests) can help you with future 
+development process. 
+Whenever you're changing your code, by adding new functionality or improving performance, 
+you can check if you're still able to reproduce your previous results.
```

Comment:
i think this should be clarified into a callout.

changing functionality may change output, which is ok. but it's important to understand what aspect of the code change created a change in output. thus regression testing can help associate changes in output with changes in code.